vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -17,29 +17,33 @@ def initialize(occupancy)
       def process
         return unless occupancy.account_id?
 
-        ActiveRecord::Base.transaction do
-          create_order
-          create_order_detail
-
-          order.validate_order!
-          order.purchase!
-        end
+        create_order unless occupancy.order_detail_id?
+        fulfill_order if occupancy.complete?
 
         order
       end
 
       private
 
       def create_order
+        ActiveRecord::Base.transaction do
+          create_order_and_detail
+          order.validate_order!
+          order.purchase!
+        end
+      end
+
+      def fulfill_order
+        order_detail.update_order_status! occupancy.user, OrderStatus.complete.first
```

Comment:
while this updates the status to be complete, I'm not sure of a good context to reference to see what else is involved. any particular suggestions?